planning/behavior_path_planner/src/utils/avoidance/utils.cpp

```diff
@@ -1367,4 +1367,181 @@ AvoidLineArray combineRawShiftLinesWithUniqueCheck(
+    rh->getLaneletSequence(current_lane, vehicle_pose, backward_distance, forward_distance);
+
+  lanelet::ConstLanelets lanes{};
+  for (const auto & lane : ego_succeeding_lanes) {
+    const auto opt_left_lane = rh->getLeftLanelet(lane);
+    if (!is_right && opt_left_lane) {
+      lanes.push_back(opt_left_lane.get());
+    }
+
+    const auto opt_right_lane = rh->getRightLanelet(lane);
+    if (is_right && opt_right_lane) {
+      lanes.push_back(opt_right_lane.get());
+    }
+
+    const auto right_opposite_lanes = rh->getRightOppositeLanelets(lane);
+    if (is_right && !right_opposite_lanes.empty()) {
+      lanes.push_back(right_opposite_lanes.front());
+    }
+  }
+
+  return lanes;
+}
+
+std::vector<ExtendedPredictedObject> getSafetyCheckTargetObjects(
+  const AvoidancePlanningData & data, const std::shared_ptr<const PlannerData> & planner_data,
+  const std::shared_ptr<AvoidanceParameters> & parameters, const bool is_right)
+{
+  const auto & p = parameters;
+  const auto check_right_lanes =
+    (is_right && p->check_shift_side_lane) || (!is_right && p->check_other_side_lane);
+  const auto check_left_lanes =
+    (!is_right && p->check_shift_side_lane) || (is_right && p->check_other_side_lane);
+
+  std::vector<ExtendedPredictedObject> target_objects;
+
+  const auto append_target_objects = [&](const auto & check_lanes, const auto & objects) {
+    std::for_each(objects.begin(), objects.end(), [&](const auto & object) {
+      if (isCentroidWithinLanelets(object.object, check_lanes)) {
+        target_objects.push_back(utils::avoidance::transform(object.object, p));
+      }
+    });
+  };
+
+  const auto unavoidable_objects = [&data]() {
+    ObjectDataArray ret;
+    std::for_each(data.target_objects.begin(), data.target_objects.end(), [&](const auto & object) {
+      if (!object.is_avoidable) {
+        ret.push_back(object);
+      }
+    });
+    return ret;
+  }();
+
+  // check right lanes
+  if (check_right_lanes) {
+    const auto check_lanes = getAdjacentLane(planner_data, p, true);
+
+    if (p->check_other_object) {
+      append_target_objects(check_lanes, data.other_objects);
+    }
+
+    if (p->check_unavoidable_object) {
+      append_target_objects(check_lanes, unavoidable_objects);
+    }
+  }
+
+  // check left lanes
+  if (check_left_lanes) {
+    const auto check_lanes = getAdjacentLane(planner_data, p, false);
+
+    if (p->check_other_object) {
+      append_target_objects(check_lanes, data.other_objects);
+    }
+
+    if (p->check_unavoidable_object) {
+      append_target_objects(check_lanes, unavoidable_objects);
+    }
+  }
+
+  // check current lanes
+  if (p->check_current_lane) {
+    const auto check_lanes = data.current_lanelets;
+
+    if (p->check_other_object) {
+      append_target_objects(check_lanes, data.other_objects);
+    }
+
+    if (p->check_unavoidable_object) {
+      append_target_objects(check_lanes, unavoidable_objects);
+    }
+  }
+
+  return target_objects;
+}
```

Comment:
just my suggestion you can ignore but I thought `is_right` is better to be changed if my understand is correct .
```suggestion
std::vector<ExtendedPredictedObject> getSafetyCheckTargetObjects(
  const AvoidancePlanningData & data, const std::shared_ptr<const PlannerData> & planner_data,
  const std::shared_ptr<AvoidanceParameters> & parameters, const bool is_shifting_right)
{
  struct CheckCondition
  {
    bool should_check;
    enum CheckLane { RIGHT_LANE, LEFT_LANE, CURRENT_LANE } check_lane;
  };

  const auto & p = parameters;
  std::vector<ExtendedPredictedObject> target_objects;

  const auto unavoidable_objects = [&data]() {
    ObjectDataArray ret;
    std::for_each(data.target_objects.begin(), data.target_objects.end(), [&](const auto & object) {
      if (!object.is_avoidable) {
        ret.push_back(object);
      }
    });
    return ret;
  }();

  // clang-format off
  CheckCondition conditions[] = {
    {  is_shifting_right && p->check_shift_side_lane || !is_shifting_right && p->check_other_side_lane, CheckCondition::RIGHT_LANE },
    { !is_shifting_right && p->check_shift_side_lane ||  is_shifting_right && p->check_other_side_lane, CheckCondition::LEFT_LANE },
    { p->check_current_lane, CheckCondition::CURRENT_LANE },
  };
  // clang-format on

  for (const auto & condition : conditions) {
    if (!condition.should_check) {
      continue;
    }

    const auto check_lanes = condition.check_lane == CheckCondition::CURRENT_LANE
                               ? data.current_lanelets
                               : (condition.check_lane == CheckCondition::RIGHT_LANE
                                    ? getAdjacentLane(planner_data, p, true)
                                    : getAdjacentLane(planner_data, p, false));

    if (p->check_other_object) {
      for (const auto & object : data.other_objects) {
        if (isCentroidWithinLanelets(object.object, check_lanes)) {
          target_objects.push_back(utils::avoidance::transform(object.object, p));
        }
      }
    }

    if (p->check_unavoidable_object) {
      for (const auto & object : unavoidable_objects) {
        if (isCentroidWithinLanelets(object.object, check_lanes)) {
          target_objects.push_back(utils::avoidance::transform(object.object, p));
        }
      }
    }
  }

  return target_objects;
}
```

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4388/commits/e72a556715d9437fa6b0c47486470569a339f22d.